### PR TITLE
Add lock flags for fcntl, flock & for idn2

### PR DESCRIPTION
**Pull request by ahmedelgabri**
As discussed here https://github.com/neomutt/neomutt/issues/1228#issuecomment-392265946

### CODE REVIEW

File: Formula/neomutt.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@ class Neomutt < Formula
   option "with-lmdb", "Build with lmdb support"
   option "with-lua", "Build with lua scripting support enabled"
   option "with-s-lang", "Build against slang instead of ncurses"
+  option "with-fcntl", "Build with fcntl support"
```
Comment (flatcap) [inline Formula/neomutt.rb]: It might be worth adding a "(default)" comment here

**Review by flatcap — COMMENTED** [review]
Looks good.

This isn't your fault, but...
The Travis build is reporting failure each time: https://travis-ci.org/neomutt/homebrew-neomutt/jobs/384294925
Something in the build script `.travis.yml` is returning 1.
If you could find out what, that'd be great.

File: Formula/neomutt.rb (hunk 1 above)
Comment (ahmedelgabri) [reply Formula/neomutt.rb]: You mean like this?

```
option "with-fcntl", "Build with fcntl support (default)"
```

**Comment by ahmedelgabri** [comment]
I'll have a look at travis & see if I can figure out this & will also add `idn2` flag after your comment here https://github.com/neomutt/neomutt/issues/1228#issuecomment-392320195

File: Formula/neomutt.rb (hunk 1 above)
Comment (flatcap) [reply Formula/neomutt.rb]: Yes.  That's the preferred option and if the user doesn't specify anything, that's what they'll get.

**Comment by ahmedelgabri** [comment]
Travis fails on `brew link python@2` command here https://travis-ci.org/neomutt/homebrew-neomutt/jobs/384098400#L4603-L4615 which seems like a dependency of dependency 

also added `idn2` flags

**Comment by flatcap** [comment]
@jindraj @sharktamer @alexpearce 

Does this look OK to you?
Do you have any ideas how to stop Travis failing?
How do you feel about adding Lua support to the upstream Formula?

**Comment by ahmedelgabri** [comment]
I can add `brew link --force python@2` but not sure if this will fix the exit code + what if the dependency changed?

**Comment by flatcap** [comment]
Ok, that might do for now.  Travis doesn't change that often.

**Comment by flatcap** [comment]
Ah, well.  Worth a try.

**Comment by ahmedelgabri** [comment]
@flatcap thanks, but I forgot to remove this commit https://github.com/neomutt/homebrew-neomutt/pull/40/commits/eee3b8d7e41b031be0d2a07ac20f751b8e005cc1 was waiting for feedback :) will open a PR to remove it since it's useless

**Comment by flatcap** [comment]
> will open a PR

There's no need.  If **you're** happy with the change, commit it directly to master.

**Pull request merged**